### ISSUE TITLE
Bug: Playground clears portions of text when attempting to type

### ISSUE THREAD

**Issue by lostfictions** [typescript-eslint/typescript-eslint#6336]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

website

### Playground Link

https://typescript-eslint.io/play

### Repro Code

```typescript
1. Visit https://typescript-eslint.io/play.
2. Begin to type in the editor. For example, I started to type `type A = { }`.
3. Portions of the text will disappear as I type. For example, after attempting to type the above, the editor seemed to delete the start of the line, and showed only `{ }`. On another attempt to type the same thing, the editor deleted everything.
```

### ESLint Config

```javascript
N/A
```

### tsconfig

```jsonc
N/A
```

### Expected Result

The playground editor should not delete or modify input.

### Actual Result

The playground editor cannot be used for entering text.

### Additional Info

Sorry to file this immediately after #6335 -- I'm not sure if it's related, or if the situation is simply much worse than I at first noticed. The playground seems pretty much unusable in this state -- I thought it would be an easy way to create a minimal repro for a rule false positive, but it's making things much more difficult.

Here's a video showing the issue. After a few moments the playground deletes the entire contents of the editor. On other attempts, different parts of the code were erased for me -- once I was left with just `{ }`, another time I had `type = `...

https://user-images.githubusercontent.com/567041/212407186-bc5ba5ac-568a-42a2-abe2-e65f756df8e2.mp4

### Versions

Chrome 108

**Comment by armano2** [comment]
note: root cause is related to https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/website/src/components/hooks/useHashState.ts
